libs/sdk-bindings/bindings-swift/Sources/BreezSDKNotification/BreezSDKConnector.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import os.log
+
+#if DEBUG && true
+fileprivate var log = Logger(
+    subsystem: Bundle.main.bundleIdentifier!,
+    category: "BreezSDKConnector"
+)
+#else
+fileprivate var log = Logger(OSLog.disabled)
+#endif
+
+class BreezSDKConnector {
+    private static var breezSDK: BlockingBreezServices? = nil
+    fileprivate static var queue = DispatchQueue(label: "BreezSDKConnector")
+    fileprivate static var sdkListener: EventListener? = nil
+    
+    static func register(connectRequest: ConnectRequest, logger: LogStream?, listener: EventListener) throws -> BlockingBreezServices? {
+        try BreezSDKConnector.queue.sync { [] in
+            BreezSDKConnector.sdkListener = listener
+            if BreezSDKConnector.breezSDK == nil {
+                BreezSDKConnector.breezSDK = try BreezSDKConnector.connectSDK(connectRequest: connectRequest, logger: logger)
+            }
+            return BreezSDKConnector.breezSDK
+        }
+    }
+    
+    static func unregister() {
+        BreezSDKConnector.queue.sync { [] in
+            BreezSDKConnector.sdkListener = nil
+        }
+    }
+    
+    static func connectSDK(connectRequest: ConnectRequest, logger: LogStream?) throws -> BlockingBreezServices? {
+        if let logger = logger {
+            try setLogStream(logStream: logger)
+        }
```

Comment:
I wonder if we can let the application call `setLogStream`? 

I notice in Kotlin the application can use `SdkLogInitializer.initializeNodeLogStream()` if they want to setup SDK logging. I assume that is optional and they can initialise it (or not) how they want. Should we do the same in Swift?

Reply:
We can, but `initializeNodeLogStream` was implemented due to a necessity on Kotlin.

**Context:**
Apps crash if `setLogStream` is called when `LogStream` is already set on `Kotlin`, but this is not an issue with `Swift` as this statement will fail and continue.
`try setLogStream(logStream: logger)`

Still, if users call `setLogger`, we're setting node log stream to that `LogStream` which may not be what all our users want.

Reply:
Removed c838429e5e47912d3700c5f130ff5f045501baaf